sklearn/decomposition/_dict_learning.py

```diff
@@ -1509,7 +1587,7 @@ class MiniBatchDictionaryLearning(_BaseSparseCoding, BaseEstimator):
     We can check the level of sparsity of `X_transformed`:
 
     >>> np.mean(X_transformed == 0)
-    0.87...
+    0.85...
```

Comment:
There's a small difference due to what I think should be considered a bug with the previous behavior. The update of the inner stats depends on the batch size see https://github.com/scikit-learn/scikit-learn/pull/18975/files#diff-20a73e7d385ab5d19a05026b635c8b256ff568e1a0e9e2fed606fec82d3b956fR1732

The thing is that due to how batches are generated, the batches may not all have the same size:
```
batches = gen_batches(n_samples, self.batch_size)
batches = itertools.cycle(batches)
```
if n_samples is not a multiple of the batches, the last batch will be smaller than batch_size. In this PR I currently just use the correct size of the batch for the update of the stats, hence the small difference.

Actually I wonder if we should do something about the generation of the batches, to make them all have the same size. wdyt ?